packages/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,285 @@
+  background-color: ${uiColors.gray.light2};
+  position: relative;
+`;
+
+const greenBorder = css`
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  background-color: ${uiColors.green.base};
+  transition: 150ms transform ease-in-out, 150ms width ease-in-out;
+`;
+
+const focusedStyle = css`
+  background-color: ${uiColors.blue.base};
+`;
+
+interface TabsProps {
+  /**
+   * Conent that will appear inside of Tabs component. Should be comprised of at least two Tabs.
+   */
+  children: Array<React.ReactElement>;
+
+  /**
+   * Callback to be executed when a tab is selected.
+   */
+  onChange?: React.ReactEventHandler;
+
+  /**
+   * Tab that should appear active. If value passed to selected prop, component will be controlled by consumer.
+   */
+  selected?: string | null;
+
+  /**
+   * className supplied to Tabs container.
+   */
+  className?: string;
+}
+
+/**
+ * # Tabs
+ *
+ * Tabs component
+ *
+ * ```
+<Tabs onChange={() => execute callback onChange}>
+  <Tab value='tab-1' title='First Tab'>Tab 1</Tab>
+  <Tab value='tab-2' title='Second Tab'>Tab 2</Tab>
+</Tabs>
+```
+ * @param props.children Content to appear inside of Tabs component.
+ * @param props.onChange Callback to be executed when a Tab is selected.
+ * @param props.selected Tab that should appear active. If value passed, component will be controlled by consumer.
+ * @param props.className Classname applied to Tabs container.
+ */
+function Tabs({ children, onChange, selected, className, ...rest }: TabsProps) {
+  const [activeTab, setActiveTab] = useState();
+  const [currentIndex, setCurrentIndex] = useState();
+  const [focusedState, setFocusedState] = useState([] as Array<string>);
+  const listRef = useRef<HTMLUListElement>(null);
+
+  useEffect(() => {
+    const currIdx = React.Children.map(
+      children,
+      (child: React.ReactElement, index) => {
+        if (child === null || child === undefined) {
+          return;
+        }
+
+        if (selected || activeTab) {
+          return child.props.value === activeTab ||
+            child.props.value === selected
+            ? index
+            : null;
+        }
+
+        if (child.props.default) {
+          return index;
+        }
+
+        return null;
+      },
+    ).filter(index => index !== null);
+
+    setCurrentIndex(currIdx[0]);
+  });
+
+  function handleChange(e: React.SyntheticEvent<Element, MouseEvent>) {
+    if (!selected) {
+      setActiveTab((e.target as HTMLLIElement).getAttribute('data-tab-id'));
+    }
+
+    if (onChange) {
+      onChange(e);
+    }
+  }
+
+  function handleKeyDown(e: React.KeyboardEvent) {
+    if (selected) {
+      return;
+    }
+
+    const enabledIndexes = React.Children.toArray(children).reduce(
+      (acc, child, index) => {
+        if (child.props.disabled) {
+          return acc;
+        }
+
+        return [...acc, index];
+      },
+      [] as Array<number>,
+    );
+
+    const enabledCurrentIndex = enabledIndexes.indexOf(currentIndex);
```

Comment:
This can just be `currentIndex`, right?

Reply:
No, because we have to make sure it's iterating through only the enabled tabs, not all of the tabs.